explainer-server/app/views/explainEditor.scala.html

```diff
@@ -68,6 +66,21 @@
             }
         }
 
+        /*
+         * This function was put here at opposition to within the scala code because
+         * the read operation of the text input uses cached values in the JS code generated by ScalaJS.
+         */
+        $(document).delegate(".explainer-editor__displayType-checkbox",'change',function() {

```

Comment:
In this case I just let scalaTags do it for me, 

```
    val checkboxTag = checkbox.render
    checkboxTag.onchange = (x: Event) => {
      g.updateCheckboxState()
    }
```

where `updateCheckboxState()` is the JavaScript function.